core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -547,7 +553,8 @@ class AdminManager(val config: KafkaConfig,
       None
     else {
       val id = resourceNameToBrokerId(resource.name)
-      if (id != this.config.brokerId)
+      // Under redirection, it is possible to handle config changes targeting at brokers other than the controller.
```

Comment:
The comment doesn't seem to make sense here. Seems like the logic doesn't have anything to do with the controller?

Reply:
The logic is needed when there is an AlterConfigRequest targeting at a specific broker. Since the non-controller node will no longer handle AlterConfigs, it is possible to see a redirected changing request with a broker.id different than the controller broker.id.